src/lib/server/postgres_message_type.hpp

```diff
@@ -50,4 +51,9 @@ enum class TransactionStatusIndicator : unsigned char {
   InFailedTransactionBlock = 'e'
 };
 
+// SQL error codes
+enum class ErrorCode : uint32_t {
+  TransactionRolledBack = 40001,
```

Comment:
A rollback could also occur as a result of the user aborting the transaction. The "error" here is TransactionConflict.

Reply:
The data type for error codes in PostgreSQL is actually five characters: https://www.postgresql.org/docs/current/errcodes-appendix.html Which code would transaction conflict be?

Reply:
Good catch. It's still 40001 (serialization_failure), but should be sent as a string.

Reply:
Let's try to support the error codes with letters from the get to. Otherwise, we need to change data types here once we add one (including potential pitfalls).

Reply:
Enum classes only support integral data types. So switching to std::map is the way to go?

Reply:
An alternative would be this enum class + a helper function (with switch case for translation). I am not sure, whether (1) performance is important here and (2) the map would hurt us. It is probably a little nicer to read since less redundant.

Reply:
I don't see us adding too many more error codes. If it makes it any easier, a constexpr char* should be fine, too.

Reply:
OK. That's OK with me, too.